### PR TITLE
Switch error state button visibility according to listener nullability

### LINKED ISSUE #90: Hide button in Error state if onRetryListener is null
The extension showError allow the developer to pass a nullable onRetryListener.
In my opinion, the retry button shouldn't appear if no listener is passed.

### PR DESCRIPTION
**Pull request by kelianClerc**
Fixes #90

### CODE REVIEW

**Review by flolom — CHANGES_REQUESTED** [review]
Please update `CHANGELOG.md`

**Review by sjcqs — COMMENTED** [review]
There are no tests at the moment but you could add some.

**Review by sjcqs — APPROVED** [review]

File: lib/src/main/ktx/com/fabernovel/statefullayout/StatefulLayout.kt [hunk 1]
```diff
@@ -75,7 +75,7 @@ fun StatefulLayout.showError(
 ): State {
     return showState(R.id.stateError).apply {
         if (errorMessage != null) setErrorMessage(errorMessage)
-        if (onRetryListener != null) setErrorRetryListener(onRetryListener)
+        setErrorRetryListener(onRetryListener)
```
Comment (flolom) [inline lib/src/main/ktx/com/fabernovel/statefullayout/StatefulLayout.kt:78]: BEFORE : You can override the error layout without providing a retry button AND pass a null `onRetryListener` by just calling the extension method `showError("error message")`

AFTER : If you do this, the `requireNotNull(retryButton)` in `private fun State.setErrorRetryListener` will be triggered and the app will crash

-> in `setErrorRetryListener` you need to check the requireNotNull only if `onRetryListener` is not null

**Review by flolom — CHANGES_REQUESTED** [review]

**Comment by sjcqs** [comment]
https://github.com/faberNovel/stateful-layout/pull/91#discussion_r492131248

As @f-lolom stated this change is incompatible with the current API. If ones needs to remove the retry button, he should override the default error state.

**Pull request closed without merging**